### PR TITLE
Imports

### PR DESCRIPTION
**Pull request by csgillespie**
* Internal: Use {yaml} instead of {config}.
  * Internal: Move {praise}, {rstudioapi}, {namer} to Suggests
  * Internal: Remove `importFrom` - be explicit

### CODE REVIEW

File: R/config.R [hunk 1]
```diff
@@ -0,0 +1,22 @@
+get_config = function() {
+  path = get_root_dir()
+  config = yaml::read_yaml(file.path(path, "notes", "config.yml"))
```
Comment (jwalton3141) [inline R/config.R:3]: I think we should now be adding `eval.expr = TRUE` to this call.

In `config.yml` for jrpyintro we have `engine.path: !expr system2("which", "python3", stdout=TRUE)`, which produces the warning:

> Warning message:
> In yaml.load(string, error.label = error.label, ...) :
>   Evaluating R expressions (!expr) will soon require explicit `eval.expr` option (see yaml.load help)

File: R/create_final.R [hunk 2]
```diff
@@ -61,38 +60,39 @@ create_final_dir = function(note_name, pracs) {
...
+  con = get_config()
   pkgs = unlist(con$deb_packages)
   return(pkgs)
 }
 
 #' @export
 #' @rdname create_final
 get_python_pkg_name = function() {
-  con = config::get()
+  con = get_config()
```
Comment (jwalton3141) [inline R/create_final.R:80]: One for another time...

But I think `get_python_pkg_name()` and `get_r_pkg_name()` functions should be renamed `get_python_pkgs()` and `get_r_pkgs()`. This better reflects what they actually do and is aligned with `get_deb_pkgs()`.

**Review by jwalton3141 — APPROVED** [review]
Looks good!

I hadn't quite appreciated how many times we were unnecessarily and repeatedly calling `config::get()`, instead of just calling at the top of any functions which need the config.

File: R/config.R (hunk 1 above)
Comment (csgillespie) [reply R/config.R:3]: Fixed. Thanks

File: R/create_final.R (hunk 2 above)
Comment (csgillespie) [reply R/create_final.R:80]: Noted for v2 of the pkg

File: R/config.R (hunk 1 above)
Comment (csgillespie) [reply R/config.R]: Fixed.

File: R/check_urls.R [hunk 3]
```diff
@@ -13,19 +12,13 @@ url_info = function(url) {
...
 
-  # Hack to detect internet connection on laptops
-  if (isTRUE(httr::http_error("https://www.google.com"))) {
-    msg_info("No internet connection - skipping URL check")
-    return(invisible(NULL))
-  }
   msg_start("Checking URLS...check_urls()")
   tokens = read_tokens()
-  urls = dplyr::filter(tokens, X1 == "url")$X3 #nolint
+  urls = dplyr::filter(tokens, .data$X1 == "url")$X3 #nolint
```
[the remark this answers is not in the document]
Comment (csgillespie) [reply R/check_urls.R:21]: Noted for v2

**Pull request merged**